lib/nudge_web/controllers/event_controller.ex

```diff
@@ -0,0 +1,50 @@
+defmodule NudgeWeb.EventController do
+  use NudgeWeb, :controller
+  alias Nudge.Sites
+  alias Nudge.Sites.Event
+
+  def new(conn, _params) do
+    changeset = Sites.change_event(%Event{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, %{
+        "event" => %{
+          "title" => title,
+          "body" => body,
+          "start_date" => start_date,
+          "end_date" => end_date,
+          "kind" => kind,
+          "respect_dnd" => dnd
+        }
+      }) do
+    IO.inspect(conn.assigns)
+
+    case Sites.create_event(%{
+           "title" => title,
+           "body" => body,
+           "start_date" => start_date,
+           "end_date" => end_date,
+           "kind" => kind,
+           "respect_dnd" => dnd,
+           "site_id" => 7
+         }) do
+      {:ok, event} ->
+        conn
+        |> put_flash(:info, "Site created successfully.")
+        |> redirect(to: Routes.site_event_path(conn, :index, :id))
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        render(conn, "new.html", changeset: changeset)
+    end
+  end
+
+  def index(conn, %{"site_id" => id}) do
+    id = 7
+    site_id = Nudge.Accounts.get_site!(id)
+
+    IO.inspect(site_id)
+    events = Sites.list_site_events(site_id)
```

Comment:
Does `get_site!/1` return just id or the whole site struct with data?

This is why we end up having type casting error in the query. The query expects and id, but you are passing the whole result to the `list_site_events/1`